### PR TITLE
Improve performance of import

### PR DESCRIPTION
**Pull request by kitaisreal**
Improve performance of import

### CODE REVIEW

File: ydb/public/lib/ydb_cli/import/import.cpp [hunk 1]
```diff
@@ -378,28 +432,51 @@ TStatus TImportFileClient::UpsertCsv(IInputStream& input, const TString& dbPath,
...
-            inFlightRequests.push_back(UpsertCsvBuffer(dbPath, buffer));
-            buffer = headerRow;
+        if (batchBytes < settings.BytesPerRequest_) {
+            continue;
+        }
+
+        if (inputSizeHint && progressCallback)
+        {
+            double progress = static_cast<double>(readBytes) / *inputSizeHint;
+            progressCallback(static_cast<size_t>(progress * 100));
```
Comment (4ertus2) [inline ydb/public/lib/ydb_cli/import/import.cpp:464]: Could we remove magic const 100 from here? I.e. we could call
progressCallback(readBytes, inputSizeHint)

File: ydb/public/lib/ydb_cli/import/import.cpp [hunk 2]
```diff
@@ -245,42 +266,70 @@ TStatus TImportFileClient::Import(const TVector<TString>& filePaths, const TStri
...
+            }
 
-            if (resultStatus != EStatus::SUCCESS) {
-                return MakeStatus(EStatus::SCHEME_ERROR,
-                    TStringBuilder() <<  result.GetIssues().ToString() << dbPath);
+            ProgressCallbackFunc progressCallback;
+
+            if (isStdoutInteractive)
+            {
+                size_t oldProgress = 0;
```
Comment (4ertus2) [inline ydb/public/lib/ydb_cli/import/import.cpp:324]: Do we have guaranty that oldProgress allocated on stack would not be reused out of if () scope?

File: ydb/public/lib/ydb_cli/import/import.cpp [hunk 3]
```diff
@@ -585,49 +691,61 @@ TStatus TImportFileClient::UpsertParquet([[maybe_unused]] const TString& filenam
...
+                        // Single row or fits into the byte limit.
+                        auto value = UpsertParquetBuffer(dbPath, NYdb_cli::NArrow::SerializeBatchNoCompression(rowsBatch), strSchema);
+                        auto status = value.ExtractValueSync();
+                        if (!status.IsSuccess())
+                            return status;
+
+                        uploadedRowsCallback(rowsBatch->num_rows());
+                    } else {
+                        // Split current slice.
+                        rowsToSendBatches.push(rowsBatch->Slice(rowsBatch->num_rows() / 2));
```
Comment (4ertus2) [inline ydb/public/lib/ydb_cli/import/import.cpp:735]: minor:

i64 halfLen = rowsBatch->num_rows() / 2;
rowsToSendBatches.push(rowsBatch->Slice(halfLen));
rowsToSendBatches.push(rowsBatch->Slice(0, halfLen));

File: ydb/public/lib/ydb_cli/import/import.cpp [hunk 4]
```diff
@@ -585,49 +691,61 @@ TStatus TImportFileClient::UpsertParquet([[maybe_unused]] const TString& filenam
...
-                    // Single row or fits into the byte limit.
-                    auto status = WaitForQueue(inFlightGetter.GetCurrentMaxInflight(), inFlightRequests);
-                    if (!status.IsSuccess()) {
-                        return status;
+                    // Nothing to send. Continue.
+                    if (rowsBatch->num_rows() == 0) {
+                        continue;
                     }
+                    // Logarithmic approach to find number of rows fit into the byte limit.
+                    if (rowsBatch->num_rows() == 1 || NYdb_cli::NArrow::GetBatchDataSize(rowsBatch) < settings.BytesPerRequest_) {
```
Comment (4ertus2) [inline ydb/public/lib/ydb_cli/import/import.cpp:725]: It's better to make SerializeBatchNoCompression() and check result data size over settings.BytesPerRequest_. Then GetBatchDataSize() is not needed.

File: ydb/public/lib/ydb_cli/import/import.cpp (hunk 2 above)
Comment (kitaisreal) [reply ydb/public/lib/ydb_cli/import/import.cpp:324]: We catch `oldProgress` in lambda by value, and then pass this lambda by value into upsert function.

File: ydb/public/lib/ydb_cli/import/import.cpp (hunk 4 above)
Comment (kitaisreal) [reply ydb/public/lib/ydb_cli/import/import.cpp:751]: If we serialize batch using `SerializeBatchNoCompression` and result size will be `>= settings.BytesPerRequest_`, then we must somehow split batch, and throw away serialized result. 
Method `GetBatchDataSize` is not heavy, so it does not lead to some performance issues.

File: ydb/public/lib/ydb_cli/import/import.cpp (hunk 3 above)
Comment (kitaisreal) [reply ydb/public/lib/ydb_cli/import/import.cpp]: Updated.

**Review by 4ertus2 — APPROVED** [review]

**Pull request closed without merging**